dipy/reconst/tests/test_dti.py

```diff
@@ -106,7 +106,14 @@ def test_tensor_model():
     tensor = from_lower_triangular(D)
     A_squiggle = tensor - (1 / 3.0) * np.trace(tensor) * np.eye(3)
     mode = 3 * np.sqrt(6) * np.linalg.det(A_squiggle / np.linalg.norm(A_squiggle))
-    evecs = np.linalg.eigh(tensor)[1]
+    evals_eigh, evecs_eigh = np.linalg.eigh(tensor)
+    # Sort according to eigen-value from large to small:
+    evecs = evecs_eigh[:, np.argsort(evals_eigh)[::-1]]
+    # Check that eigenvalues and eigenvectors are properly sorted through
+    # that previous operation:
+    for i in range(3):
+        assert_array_equal(np.dot(tensor, evecs[:, i]),
```

Comment:
In general, we should avoid comparing floating point values for equality, because even if the operations are equivalent, the result will in general depend on the order of evaluation, for example:
```
x = (np.float64(3.1) / np.float64(2.1)) * (np.float64(2.1) / np.float64(3.1))
y = (np.float64(3.1) * np.float64(2.1)) / (np.float64(3.1) * np.float64(2.1))
x == y
```
Which evaluates to False.
I think `assert_array_almost_equal` with a precision of about `1e-9` would be better.